hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -894,6 +1009,19 @@ def __repr__(self):
         indices=indices,
     )
 
+    if api_verson_gt(api_version, "2021.12"):
+
+        @defines_strategy()
+        def complex_dtypes(
+            *, sizes: Union[int, Sequence[int]] = (64, 128)
+        ) -> st.SearchStrategy[DataType]:
+            return _complex_dtypes(xp, sizes=sizes)
+
+        complex_dtypes.__doc__ = _complex_dtypes.__doc__
+        kwargs["complex_dtypes"] = complex_dtypes
+
+    return PrettySimpleNamespace(**kwargs)
```

Comment:
Since we already have a custom SimpleNamespace subclass, I'd be tempted to give it a complex_dtypes property which raises a more helpful AttributeError on older versions - including 'you attempted to access X on version Y, but it was added in version Z' or similar.

Reply:
I think generally working the custom subclass is nice, so I've made `name` an attribute along with `api_version`, which gets referenced in `__repr__()` and is checked for in `__init__()`.

In regards to a `complex_dtypes` property, I've come up with a solution which seems appropriate. It's kinda weird to deal with `SimpleNamespace` in this regard, but probably best to mess around with a property than to not use it. The message right now could be more helpful (i.e. it was added in x version), but awkward right now with it being a draft feature. Lemme know what you think!